chore(github): Add template for issues and pull requests (#45)
remoteoss/json-schema-form: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md

# Pull Request

## Summary

Provide a concise and clear summary of your pull request.

> ## Motivation

Reviewer: suggestion: I believe the "Motion" isn't needed in favor of the Related Issues/PRs which by itself contains the reason behind the PR. What do you think?

Author: Feels the same now. Makes it much more minimal.

Explain the motivation behind this pull request. What problem does it solve or feature does it introduce?

## Changes Made

Describe the changes made in this pull request. Be specific about code changes, new features, or bug fixes.

## Related Issues/PRs

Link to any related issues or pull requests, if applicable.

## Screenshots (Before/After if applicable)

Include screenshots or visual aids to demonstrate the changes visually. If your PR modifies the UI or behavior, include before and after descriptions or screenshots.

### Desktop

### Mobile

> ### iPad

Reviewer: Suggestion: No need for these Desktop / mobile / iPad sections because this tool is headless (no UI provided). :)

Author: Just went with the flow of standard PR templates in frontend projects 😅
Removed 👍🏼

## Checklist

> - [x] I have read the [Contributing Guidelines](../../CONTRIBUTING.md) and followed them.
> - [ ] I have tested my changes thoroughly.
> - [ ] My code follows the project's coding standards and style guide.
> - [ ] I have updated documentation (if applicable).

Reviewer: ```suggestion
- [x] I have read the [Contributing Guidelines](../../CONTRIBUTING.md) and followed them.
- [ ] Unit tests were added.
```

- I removed the "coding standards" part because it's already covered by eslint and prettier on CI (automated check)
- I removed the documentation because currently, that lives outside the repo as explained in the Contributing file.

Author: I have made the suggested changes.

> ## Any Other Context

Reviewer: suggestion: I find this section not very useful. Personally, I'd expect the authors to add this directly in the "changes made" or as part of the PR comments. What do you think?
